### PR TITLE
[node-agent] Refactor `files` handling in `OperatingSystemConfig` API

### PR DESCRIPTION
**Pull request by oliver-goetz**
**How to categorize this PR?**

/area usability
/kind enhancement

**What this PR does / why we need it**:
This PR refactors the handling of `files` in `OperatingSystemConfig` API. 

PR #8707 added a `files` section to each `unit` in order to identify files which should restart the corresponding unit when they change.
With this PR the `spec.units.files` section is replaced by a `units.filePaths` section. `filePaths` are a slice of strings which point to a `file`. The files are maintained in `spec.files`.
Validation ensures that there is always a `file` whose `path` matches with each `filePath`.

The new approach simplifies the migration from `cloud-config-download` to `node-agent`.
Thanks @timebertt for the idea 😄 

**Which issue(s) this PR fixes**:
Part of #8023

**Special notes for your reviewer**:
/cc @rfranzke 

**Release note**:

```other operator
NONE
```

### CODE REVIEW

File: pkg/apis/extensions/v1alpha1/types_operatingsystemconfig.go [hunk 1]
```diff
@@ -239,7 +236,6 @@ type OperatingSystemConfigStatus struct {
 	Units []string `json:"units,omitempty"`
 	// Files is a list of file paths that are part of the generated Cloud Config and shall be
 	// written to the host's file system.
-	// TODO(rfranzke): Deprecate this field once UseGardenerNodeAgent feature gate is promoted to GA.
```
Comment (rfranzke) [inline pkg/apis/extensions/v1alpha1/types_operatingsystemconfig.go]: Why did you remove this TODO statement? I think it's still relevant, isn't it?

File: extensions/pkg/controller/operatingsystemconfig/bash.go [hunk 2]
```diff
@@ -54,7 +54,7 @@ mkdir -p "` + path.Dir(file.Path) + `"
 
 // UnitsToDiskScript is a utility function which generates a bash script for writing the provided units and their
 // drop-ins to the disk.
-func UnitsToDiskScript(ctx context.Context, reader client.Reader, namespace string, units []extensionsv1alpha1.Unit) (string, error) {
+func UnitsToDiskScript(units []extensionsv1alpha1.Unit) (string, error) {
```
Comment (rfranzke) [inline extensions/pkg/controller/operatingsystemconfig/bash.go]: This never returns an `error`, you could remove it from the signature.

**Review by rfranzke — CHANGES_REQUESTED** [review]

File: pkg/apis/extensions/v1alpha1/types_operatingsystemconfig.go (hunk 1 above)
Comment (oliver-goetz) [reply pkg/apis/extensions/v1alpha1/types_operatingsystemconfig.go]: You're right. I mixed it up with `spec.files`  😅

**Review by rfranzke — COMMENTED** [review]
/lgtm
/approve
